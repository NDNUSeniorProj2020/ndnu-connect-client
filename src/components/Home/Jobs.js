import React from 'react';
import 'antd/dist/antd.css';
import { Card } from 'antd';
import "./Jobs.css"

class Jobs extends React.Component {
    render() {
        return (
            <div>
                <Card title="Jobs & Interships">
                    <Card type="inner" title="Communication Designer - Instagram" extra={<p>San Franciso, CA</p>}>
                        The Communications Designer will help us deliver on our mission by developing visual identities and brand systems, interior office design concepts, as well as design curation for digital and video surfaces. The ideal candidate has the ability to execute in a variety of visual styles, can articulate design decisions, iterates quickly, and collaborates seamlessly with creative leads, art directors, and content strategists. The designer has the ability to work independently from conception to fabrication, collaborating closely with the AD.
                    </Card>
                    <Card type="inner" title="Conversation Designer - Assistant" extra={<p>San Mateo, CA</p>}>
                        We are looking for an experienced and talented Conversation Designer to conceptualize, illustrate, build and deliver innovative conversational and natural language interaction paradigms. You will be expected to drive these conversational experiences from concepts and storyboards to product launch with attention to detail through elegant probabilistic designs that allow humans to interact more naturally with our devices. A positive and collaborative attitude is required, and weâ€™re looking for a self-motivated problem solver with impeccable communication skills.
                    </Card>
                    <Card type="inner" title="QA Manager" extra={<p>Belmont, CA</p>}>
                        The Quality Assurance (QA) Manager is responsible for driving the QA program across multiple titles and platforms. You will have years of project management experience in video game development, as well as experience working with, and managing, a third party QA vendor. Working directly with our vendors, you will continue building the relationship and establish QA processes for new content launches, as well as support content for existing hardware in the market, be involved in the creation and execution of test plans and assist in defining project metrics to measure our success. You will work to improve the effectiveness of the QA program by reviewing current practices and incorporating new QA processes.
                    </Card>
                </Card>,
            </div>
        );
    }
}

export default Jobs;
