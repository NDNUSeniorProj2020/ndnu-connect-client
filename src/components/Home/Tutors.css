
.tutors-card-container {
	width: 350px;
	margin: 15px;
}

.tutors-card-title {
	font-size: 14;
    color: rgba(0, 0, 0, 0.85);
    margin-bottom: 16;
    font-weight: 500;
}

.stars-container {
	padding-left: 40px;
}

.row-style {
	text-align: left;
	height: 45px;
}
