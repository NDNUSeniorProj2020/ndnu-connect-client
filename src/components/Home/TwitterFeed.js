import React, { useEffect } from 'react';
import { Card } from 'antd';

export default function TwitterFeed() {
	useEffect(() => {
		const script = document.createElement("script");
		script.src = 'https://platform.twitter.com/widgets.js';
		document.getElementsByClassName("twitter-embed")[0].appendChild(script);
	}, []);

	return (
		<Card className="tutors-card-container">
			<section className={'twitter-container'}>
				<div className={'twitter-embed'}>
					<a className="twitter-timeline" data-width="400" data-height="400" data-dnt="true"
						href="https://twitter.com/NotreDameCA?ref_src=twsrc%5Etfw">Tweets by NotreDameCA</a>
				</div>
			</section>
		</Card>
	);
}
