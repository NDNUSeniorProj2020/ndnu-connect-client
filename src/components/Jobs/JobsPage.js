import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Jobs.css';
import FilterJobsContainer from './FilterJobsContainer';
import { fetchJobs } from '../../actions/jobs/jobsActions';

export function ConnectedJobsPage({ jobs, success, fetchJobs }) {
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const token = localStorage.getItem('token');

    fetchJobs(token);
    setIsLoading(false);
  }, [isLoading, fetchJobs]);

  if (success && !isLoading)
    return <FilterJobsContainer jobs={jobs} />;
  if (!success && !isLoading)
    return <p style={{ color: 'red' }}>Cannot load jobs.</p>;

  return <p>Loading page...</p>;
}

ConnectedJobsPage.propTypes = { jobs: PropTypes.array, success: PropTypes.bool, fetchJobs: PropTypes.func };
ConnectedJobsPage.defaultProps = { jobs: [], success: false, fetchJobs: f => f };

const mapStateToProps = ({ jobsReducer }) => ({ jobs: jobsReducer.jobs, success: jobsReducer.success });
const JobsPage = connect(mapStateToProps, { fetchJobs })(ConnectedJobsPage);

export default JobsPage;
