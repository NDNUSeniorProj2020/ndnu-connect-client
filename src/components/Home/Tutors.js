import React from 'react';
import 'antd/dist/antd.css';
import { Avatar, Card, Divider, Icon, Tag } from 'antd';
import "./Tutors.css"

const { Meta } = Card;

class Tutors extends React.Component {
  // constructor(props) {
  //   super(props);
  // }

  render() {
    const data = [
    {
      fullName: 'John Smith',
      subjects: ["Math", "Computer Science"],
      image: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png",
      rating: 5
    }, {
      fullName: 'Jane Williams',
      subjects: ["English", "History", "Humanities"],
      image: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png",
      rating: 5
    }, {
      fullName: 'James Miller',
      subjects: ["Biology", "Chemistry"],
      image: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png",
      rating: 4
    }, {
      fullName: 'Will Johnson',
      subjects: ["Business", "Accounting"],
      image: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png",
      rating: 4
    }, {
      fullName: 'Emma Jones',
      subjects: ["Psychology", "Religion"],
      image: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png",
      rating: 4
    }
    ];
    return (
      <div>
        <Card className="tutors-card-container">
        <p className="tutors-card-title">
          Top Tutors
        </p>
        {
          data.map(tutor => {
            let namePlusStars = (
              <div>
                {tutor.fullName}
                <span className="stars-container">
                  {Array.from(Array(tutor.rating)).map((star, i) => <Icon type="star" theme="twoTone" key={i} twoToneColor="#FFD700" />)}
                </span>
              </div>
            )
            return(
              <a key={tutor.fullName} href="google.com">
                <Meta
                  avatar={
                    <Avatar src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png" />
                  }
                  title={namePlusStars}
                  className="row-style"
                  description={
                    tutor.subjects.map(subject => <Tag key={subject}>{subject}</Tag>)
                  }
                />
                <Divider />
              </a>
            )
          })
        }
        </Card>
      </div>
    );
  }
}

export default Tutors;
          