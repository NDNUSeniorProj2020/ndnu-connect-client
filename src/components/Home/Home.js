import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Home.css';
import TwitterFeed from './TwitterFeed';
import TutorsWidget from '../Tutors/TutorsWidget';
import JobsWidget from '../Jobs/JobsWidget';
import { fetchJobs } from '../../actions/jobs/jobsActions';
import { fetchTutors } from '../../actions/tutor/tutorActions';
import { Row, Col, Card } from 'antd';

export function ConnectedHome({ jobs, tutors, success, fetchTutors, fetchJobs }) {
	const [isLoading, setIsLoading] = useState(true);

	useEffect(() => {
		const token = localStorage.getItem('token');

		fetchTutors(token);
		fetchJobs(token);
		setIsLoading(false);
	}, [isLoading, fetchTutors, fetchJobs]);

	if (success && !isLoading)
		return (
			<div>
				<Row gutter={16}>
					<Col span={16}>
						<JobsWidget jobs={jobs} />
					</Col>
					<Col span={8}>
						<Row gutter={16}>
							<TwitterFeed />
						</Row>
						<Row gutter={16}>
							<TutorsWidget />
						</Row>
					</Col>
				</Row>
			</div>
		);

	return <h1>Dashboard</h1>;
}

ConnectedHome.propTypes = {
	boards: PropTypes.array,
	jobs: PropTypes.array,
	tutors: PropTypes.array,
	fetchTutors: PropTypes.func,
	fetchJobs: PropTypes.func,
};

ConnectedHome.defaultProps = {
	jobs: [],
	tutors: [],
	fetchTutors: f => f,
	fetchJobs: f => f,
};

const mapStateToProps = ({ jobsReducer, tutorReducer }) => ({
	jobs: jobsReducer.jobs,
	tutors: tutorReducer.tutors,
	success: jobsReducer.success && tutorReducer.success
});
const Home = connect(mapStateToProps, { fetchTutors, fetchJobs })(ConnectedHome);

export default Home;
