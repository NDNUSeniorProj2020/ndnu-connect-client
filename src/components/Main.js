import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Home from './Home/Home';

export default function Main() {
	return (
		<div>
			<Router>
				<Switch>
					<Route path="/">
						<Home />
					</Route>
				</Switch>
			</Router>
		</div>
	);
}