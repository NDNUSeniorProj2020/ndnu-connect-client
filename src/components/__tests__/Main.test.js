import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter, Route } from 'react-router-dom';

import Main from '../Main';
import Home from '../Home/Home';
import JobsPage from '../Jobs/JobsPage';

describe('tests for Main', () => {
	describe('snapshot tests', () => {
		it('renders without crashing', () => {
			const tree = shallow(<BrowserRouter><Main /></BrowserRouter>);
			expect(tree).toMatchSnapshot();
		});
	});

	describe('integration tests', () => {
		it('tests React Router routes', () => {
			const wrapper = shallow(<Main />);
			const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
				const routeProps = route.props();
				pathMap[routeProps.path] = routeProps.component;
				return pathMap;
			}, {});

			expect(pathMap['/jobs']).toBe(JobsPage);
			expect(pathMap['/']).toBe(Home);
		});
	});
});
