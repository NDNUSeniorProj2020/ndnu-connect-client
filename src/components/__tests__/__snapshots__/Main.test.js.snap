// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Main snapshot tests renders without crashing 1`] = `ShallowWrapper {}`;
